src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private String filepath;
+    private String path;
+
+    /**
+     * Constructor of Storage object.
+     * @param filepath filepath of duke.txt.
+     * @param path path of the folder to store the file.
+     */
+    public Storage(String filepath, String path) {
+        this.filepath = filepath;
+        this.path = path;
+    }
+
+    /**
+     * Opens the file on hard disk where user's past tasks were saved.
+     * Loads the previous tasks saved in duke.txt upon start.
+     * @return list of tasks to be used to construct TaskList.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Path directoryPath = Paths.get(path);
+            Files.createDirectories(directoryPath); //Create directory if it does not exist
+            File taskRecord = new File(filepath);
+            if (!taskRecord.createNewFile()) {
+                //File already exists, parse file and add duke.tasks to ArrayList
```

Comment:
```suggestion
// File already exists, parse file and add duke.tasks to ArrayList
```
I think inline comments look better with a space first.